lib/services/datasets.js

```diff
@@ -0,0 +1,528 @@
+'use strict';
+
+var invariant = require('invariant'),
+  geojsonhint = require('geojsonhint/object'),
+  request = require('superagent'),
+  hat = require('hat'),
+  makeService = require('../make_service'),
+  constants = require('../constants'),
+  makeURL = require('../make_url');
+
+var Datasets = module.exports = makeService('MapboxDatasets');
+
+/**
+ * To retrieve a listing of datasets for a particular account.
+ * This request requires an access token with the datasets:read scope.
+ *
+ * @param {Function} callback called with (err, datasets)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ * mapboxClient.listDatasets(function(err, datasets) {
+ *   console.log(datasets);
+ *   // [
+ *   //   {
+ *   //     "owner": {account},
+ *   //     "id": {dataset id},
+ *   //     "name": {dataset name},
+ *   //     "description": {dataset description},
+ *   //     "created": {timestamp},
```

Comment:
I'm 100% with you on the ideal case here. But `@returns` just isn't the right place to put this. What you really want to do is document the `@param {Function} callback`, which is a part of JSDoc that seems kinda clumsy and isn't supported in documentationjs. 

I do think this is something we ought to elevate to the shortlist in documentationjs: https://github.com/documentationjs/documentation/issues/9